go.sum

```diff
@@ -0,0 +1,286 @@
+github.com/mitchellh/colorstring v0.0.0-20150917214807-8631ce90f286/go.mod h1:l0dey0ia/Uv7NcFFVbCLtqEBQbrT4OCwCSKTEv6enCw=
+github.com/mitchellh/copystructure v0.0.0-20170525013902-d23ffcb85de3 h1:dECZqiJYhKdj9QlLpiQaRDXHDXRTdiyZI3owdDGhlYY=
+github.com/mitchellh/copystructure v0.0.0-20170525013902-d23ffcb85de3/go.mod h1:eOsF2yLPlBBJPvD+nhl5QMTBSOBbOph6N7j/IDUw7PY=
+github.com/mitchellh/go-homedir v0.0.0-20161203194507-b8bc1bf76747 h1:eQox4Rh4ewJF+mqYPxCkmBAirRnPaHEB26UkNuPyjlk=
+github.com/mitchellh/go-homedir v0.0.0-20161203194507-b8bc1bf76747/go.mod h1:SfyaCUpYCn1Vlf4IUYiD9fPX4A5wJrkLzIz1N1q0pr0=
+github.com/mitchellh/go-linereader v0.0.0-20141013185533-07bab5fdd958 h1:wN+5lV34eSnVSZgBLWRGHr6L4giR3/wI2B9DLmVnlfI=
+github.com/mitchellh/go-linereader v0.0.0-20141013185533-07bab5fdd958/go.mod h1:OaY7UOoTkkrX3wRwjpYRKafIkkyeD0UtweSHAWWiqQM=
+github.com/mitchellh/go-testing-interface v0.0.0-20170730050907-9a441910b168 h1:FW/lWFII8EehRx+hVNy5OkkIhWXz9NC69vO5Zr2RExY=
+github.com/mitchellh/go-testing-interface v0.0.0-20170730050907-9a441910b168/go.mod h1:kRemZodwjscx+RGhAo8eIhFbs2+BFgRtFPeD/KE+zxI=
+github.com/mitchellh/go-wordwrap v0.0.0-20150314170334-ad45545899c7 h1:DpOJ2HYzCv8LZP15IdmG+YdwD2luVPHITV96TkirNBM=
+github.com/mitchellh/go-wordwrap v0.0.0-20150314170334-ad45545899c7/go.mod h1:ZXFpozHsX6DPmq2I0TCekCxypsnAUbP2oI0UX1GXzOo=
+github.com/mitchellh/hashstructure v0.0.0-20160209213820-6b17d669fac5 h1:h+4fp6yIoLPf/K2egDK3kvYM2zqb28gJIWWMiDzBdKM=
+github.com/mitchellh/hashstructure v0.0.0-20160209213820-6b17d669fac5/go.mod h1:QjSHrPWS+BGUVBYkbTZWEnOh3G1DutKwClXU/ABz6AQ=
+github.com/mitchellh/mapstructure v0.0.0-20170307201123-53818660ed49 h1:kaWdlw4YogwkDl8CG+/VxhXkrL9uz3n1D9QBC2pEGLE=
+github.com/mitchellh/mapstructure v0.0.0-20170307201123-53818660ed49/go.mod h1:FVVH3fgwuzCH5S8UJGiWEs2h04kUh9fWfEaFds41c1Y=
+github.com/mitchellh/panicwrap v0.0.0-20161208170302-ba9e1a65e0f7 h1:+PBI9A4rLQJlch3eQI/RkTY2HzX+bl2lPUf3goenBxs=
+github.com/mitchellh/panicwrap v0.0.0-20161208170302-ba9e1a65e0f7/go.mod h1:QuAqW7/z+iv6aWFJdrA8kCbsF0OOJVKCICqTcYBexuY=
+github.com/mitchellh/prefixedio v0.0.0-20151214002211-6e6954073784 h1:+DAetXqxv/mSyCkE9KBIYOZs9b68y7SUaDCxQMRjA68=
+github.com/mitchellh/prefixedio v0.0.0-20151214002211-6e6954073784/go.mod h1:kB1naBgV9ORnkiTVeyJOI1DavaJkG4oNIq0Af6ZVKUo=
+github.com/mitchellh/reflectwalk v0.0.0-20170726202117-63d60e9d0dbc h1:gqYjvctjtX4GHzgfutJxZpvZ7XhGwQLGR5BASwhpO2o=
+github.com/mitchellh/reflectwalk v0.0.0-20170726202117-63d60e9d0dbc/go.mod h1:mSTlrgnPZtwu0c4WaC2kGObEpuNDbx0jmZXqmk4esnw=
+github.com/nu7hatch/gouuid v0.0.0-20131221200532-179d4d0c4d8d h1:VhgPp6v9qf9Agr/56bj7Y/xa04UccTW04VP0Qed4vnQ=
+github.com/nu7hatch/gouuid v0.0.0-20131221200532-179d4d0c4d8d/go.mod h1:YUTz3bUH2ZwIWBy3CJBeOBEugqcmXREj14T+iG/4k4U=
+github.com/oklog/run v1.0.0 h1:Ru7dDtJNOyC66gQ5dQmaCa0qIsAUFY3sFpK1Xk8igrw=
+github.com/oklog/run v1.0.0/go.mod h1:dlhp/R75TPv97u0XWUtDeV/lRKWPKSdTuV0TZvrmrQA=
```

Comment:
It looks like this dependency is causing some issues for `go mod tidy`:

```
go: finding github.com/dnaeon/go-vcr v0.0.0-20170218072653-87d4990451a8
go: github.com/oklog/run@v1.0.0: unknown revision v1.0.0
go: finding github.com/Azure/azure-sdk-for-go v10.3.0-beta+incompatible
```

Do you mind `rm go.sum` and re-running `go mod tidy`?

Reply:
How are you able to get that? Hmm makes me think modules still has some inconsistent behavior

Reply:
```
~/git/go/src/github.com/terraform-providers/terraform-provider-huaweicloud
❯ GO111MODULE=on go mod tidy
go: finding github.com/smartystreets/goconvey v0.0.0-20190222223459-a17d461953aa
go: finding github.com/huaweicloud/golangsdk v0.0.0-20190218081629-94fd0333a106
go: finding github.com/hashicorp/terraform v0.11.12-beta1.0.20190227065421-fc531f54a878
go: finding github.com/hashicorp/go-tfe v0.3.10
go: finding golang.org/x/sys v0.0.0-20190228124157-a34e9553db1e
go: finding gopkg.in/yaml.v2 v2.0.0-20170812160011-eb3733d160e7
go: downloading github.com/hashicorp/terraform v0.11.12-beta1.0.20190227065421-fc531f54a878
go: downloading github.com/huaweicloud/golangsdk v0.0.0-20190218081629-94fd0333a106
go: downloading gopkg.in/yaml.v2 v2.0.0-20170812160011-eb3733d160e7
go: downloading github.com/smartystreets/goconvey v0.0.0-20190222223459-a17d461953aa
go: downloading golang.org/x/sys v0.0.0-20190228124157-a34e9553db1e

~/git/go/src/github.com/terraform-providers/terraform-provider-huaweicloud 8s
❯ git status
On branch go-modules-2019-03-01
nothing to commit, working tree clean
```

Reply:
I really don't like that when I run `go mod tidy` with the go.sum existing, it is unchanged, but delete it and re-run it changes...

Reply:
> I really don't like that when I run go mod tidy with the go.sum existing, it is unchanged, but delete it and re-run it changes...

That is odd indeed. I assume there are still some bugs in go modules which get triggered in some edge-case-y, hard-to-reproduce situations when something happens in the cache. 😞 

Either way I always make sure that I test in a clean environment without any cache etc. when I test these PRs, specifically in `golang:1.11.5` in Docker. The only downside of this approach is just long wait until all dependencies get downloaded from scratch inside the container.